backend/asset/timedmetadata.images.go

```diff
@@ -0,0 +1,76 @@
+package asset
+
+import (
+	"context"
+	"os"
+	"path"
+	"sync"
+
+	"github.com/ansel1/merry/v2"
+	"github.com/bcc-code/bcc-media-platform/backend/files"
+	"github.com/bcc-code/mediabank-bridge/log"
+	"golang.org/x/sync/errgroup"
+)
+
+// ingestImagesFromS3 downloads images from S3 and uploads them to the platform
+func ingestImagesFromS3(ctx context.Context, imagePaths []string, services externalServices, config config) (map[string]string, error) {
+	s3client := services.GetS3Client()
+	tempDir, err := os.MkdirTemp(config.GetTempDir(), "timedmetadata")
+	if err != nil {
+		return nil, merry.Wrap(err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	var eg errgroup.Group
+	imageIDs := make(map[string]string)
+	var mu sync.Mutex
+	for _, image := range imagePaths {
+		i := image
+		eg.Go(func() error {
+			localPath := path.Join(tempDir, i)
+			_, err := downloadFromS3(ctx, downloadFromS3Params{
+				client:    s3client,
+				bucket:    config.GetIngestBucket(),
+				path:      i,
+				localPath: localPath,
+			})
+			if err != nil {
+				return merry.Wrap(err)
+			}
+
+			imageId, err := uploadFileToPlatform(ctx, services, localPath)
+			if err != nil {
+				return merry.Wrap(err)
+			}
+			mu.Lock()
+			imageIDs[i] = *imageId
+			mu.Unlock()
+			return nil
```

Comment:
Is there a direct need to download the file to disk? It seems like the stream could be passed directly from the "dowload" to the "upload" functions.

Reply:
nope. changed to do that